core/account-id/src/lib.rs

```diff
@@ -57,7 +57,7 @@ impl AccountId {
 
             (!last_char_is_separator)
                 .then(|| ())
-                .ok_or(ParseAccountError(ParseErrorKind::Invalid, account_id.to_string()))
```

Comment:
I'd just 

```rust
if !last_char_is_separator {
    return Err(ParseAccountError(ParseErrorKind::Invalid, account_id.to_string()))
} 
Ok(())
```

The current version reads like overly-clever code to me.

Reply:
Ok, I actually feel strongly that we should rewrite this. I am looking at `(!last_char_is_separator).then(|| ())` and I don't actually understand whether trailing separator is an error or not. As you see from the above suggestion, I actually got this wrong. 

The correct version is (please double check)

```rust
if last_char_is_separator {
    return Err(ParseAccountError(ParseErrorKind::Invalid, account_id.to_string()))
} 
Ok(())
```